storage_manager_new.py

```diff
@@ -14,28 +14,43 @@ def clear_storage_items_and_load_items_from_file(self):
         File = open(Storage.database_file, 'r')
         for line in File.readlines():
             line = line.rstrip('\n')
-            self.items.append(Item.create_instance_from_string(line))
+            if line.startswith('Sound'):
+                self.items.append(Sound_item.create_instance_from_string(line))
+            elif line.startswith('Light'):
+                self.items.append(Light_item.create_instance_from_string(line))
+            else:
+                self.items.append(Item.create_instance_from_string(line))
 
     def write_to_database_file(self, stringified_item):
         File = open(self.database_file, 'a')
         File.write(stringified_item + '\n')
         File.close()
 
     def add_item(self, stringified_new_item_data):
+        new_item_data = stringified_new_item_data.split(
+            ', ')
         try:
-            new_item_data = stringified_new_item_data.split(
-                ', ')
             quantity = int(new_item_data.pop(3))
+        except (IndexError, ValueError) as e:
+            return e
+
+        check_value = self._check_if_value_type_is_proper(new_item_data)
+        if check_value != ValueError:
             new_item_data_without_quantity = ', '.join(
                 new_item_data)
             for i in range(1, (quantity + 1)):
-                Item.create_instance_from_string(
-                    new_item_data_without_quantity)
                 self.write_to_database_file(new_item_data_without_quantity)
             self.clear_storage_items_and_load_items_from_file()
             return True
-        except (IndexError, ValueError) as e:
-            return e
+        else:
+            return ValueError
+
+    def _check_if_value_type_is_proper(self, new_item_data):
```

Comment:
Could you add some comment or method description what it does? It's a bit unclear.